plugins/adrv9002.c

```diff
@@ -927,6 +1189,1009 @@ static void load_profile(GtkFileChooser *chooser, gpointer data)
+
+	gtk_widget_show(GTK_WIDGET(chooser));
+}
+
+static void profile_gen_update_orx(GtkComboBox* self, struct plugin_private *priv)
+{
+	int chann;
+	char widget_str[25];
+	bool tdd_en = atoi(gtk_combo_box_get_active_id(GTK_COMBO_BOX(gtk_builder_get_object(priv->builder, "cb_radio_duplex")))) == 0;
+	bool tx_en;
+
+	for (chann = 0; chann < 2; chann ++) {
+		sprintf(widget_str, "cb_tx_chan%d_en", chann + 1);
+		tx_en = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)));
+
+		sprintf(widget_str, "frame_radio_orx%d", chann + 1);
+		gtk_widget_set_sensitive(GTK_WIDGET(gtk_builder_get_object(priv->builder, widget_str)), tx_en && tdd_en);
+
+		// deactivate if disabled
+		if (!(tx_en && tdd_en)) {
+			sprintf(widget_str, "cb_radio_orx%d_en", chann + 1);
+			gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(gtk_builder_get_object(priv->builder, widget_str)), false);
+		}
+	}
+}
+
+
+char *extract_value_between(const char *str, const char *begin, const char *end)
+{
+	// if end is empty it will stop at the end of str
+	const char *i1 = strstr(str, begin);
+	if(i1 != NULL)
+	{
+		const size_t pl1 = strlen(begin);
+		const char *i2 = strcmp(end, "") != 0? strstr(i1 + pl1, end): NULL;
+		const size_t mlen = strcmp(end, "") != 0? i2 - (i1 + pl1): (int)(strlen(i1) - pl1);
+		char *ret = malloc(mlen + 1);
+
+		if(ret != NULL)
+		{
+			memcpy(ret, i1 + pl1, mlen);
+			ret[mlen] = '\0';
+			return ret;
+		}
+	}
+
+	return NULL;
+}
+
+static int profile_gen_config_get_from_device(struct adrv9002_config *cfg, gpointer data)
+{
+	struct adrv9002_config default_cfg = lte_lvs_3072_MHz_10();
+	struct plugin_private *priv = data;
+	int ret_value = 0;
+
+	// radio_config
+	radio_config radio_config;
+
+	char buf[1024];
+	char profile_config[512];
+	ret_value = iio_device_attr_read(priv->adrv9002, "profile_config", profile_config, sizeof(profile_config));
+	if (ret_value < 0) {
+		printf("\nFailed to get device attr read %s! error code: %d", "profile_config", ret_value);
+		goto iio_error;
+	}
+
+	// radio_config.ssi_lanes
+	char *ssi_interface = extract_value_between(profile_config, "SSI interface:", "");
+	if (ssi_interface == NULL) {
+		printf("\nFailed to get SSI interface!");
+		ret_value = -1;
+		goto iio_error;
+	} else if (strstr(ssi_interface, "CMOS/LVDS") != NULL) {
+		radio_config.ssi_lanes = 1;
+	} else if (strstr(ssi_interface, "LVDS") != NULL) {
+		radio_config.ssi_lanes = 2;
+	} else if (strstr(ssi_interface, "CMOS") != NULL) {
+		radio_config.ssi_lanes = 4;
+	} else {
+		printf("\nFailed to get SSI interface! got '%s' instead", ssi_interface);
+		goto iio_error;
+	}
+
+	// radio_config.ddr
+	radio_config.ddr = default_cfg.radio_cfg.ddr; // TODO
+
+	// radio_config.short_strobe
+	radio_config.short_strobe = default_cfg.radio_cfg.short_strobe; // TODO
+
+	// radio_config.lvds
+	radio_config.lvds = radio_config.ssi_lanes == 2;
+
+	// radio_config.adc_rate_mode
```

Comment:
Avoid comments like this. The names of the variables are already self explanatory